src/eos-updater-fetch.c

```diff
@@ -72,6 +74,31 @@ update_progress (OstreeAsyncProgress *progress,
     eos_updater_set_downloaded_bytes (updater, bytes);
 }
 
+static gboolean
+repo_pull (OstreeRepo *self,
+           const gchar *remote_name,
+           const gchar *ref,
+           const gchar *url_override,
+           OstreeAsyncProgress *progress,
+           GCancellable *cancellable,
+           GError **error)
+{
+  g_auto(GVariantBuilder) builder = { { { 0, } } };
+  g_autoptr(GVariant) options = NULL;
+
+  g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{sv}"));
+  g_variant_builder_add (&builder, "{s@v}", "refs",
+                         g_variant_new_variant (g_variant_new_strv (&ref, 1)));
+  if (url_override != NULL)
+    g_variant_builder_add (&builder, "{s@v}", "override-url",
+                           g_variant_new_variant (g_variant_new_string (url_override)));
+
+  options = g_variant_ref_sink (g_variant_builder_end (&builder));
+  return ostree_repo_pull_with_options (self, remote_name, options,
+                                        progress, cancellable, error);
```

Comment:
OK, I think I confused myself about the summary vs eos-summary handling when pulling from eos-update-server. I think you're right here and nothing is needed.

I'd just remove the commits and subsequent reverts. Since this code isn't pushed yet, that just adds noise to the history, IMO.